### ISSUE TITLE
More detailed git tutorials

### ISSUE THREAD

**Issue by veracioux** [veracioux/tuterm-collection#2]
The existing git tutorial could use some more work. The following aspects should be covered:

- basic coverage of branch, checkout
- basic fetching, pulling, merging and pushing
- most useful `git log` options

It would be good to add detailed tutorials/demos for each of the above functions in separate script files. In addition, `git stash`, `git rebase` should be covered in separate tuterm scripts.

**Comment by bhantsi** [comment]
Glad to know, please add hacktoberfest-accepted label then assign this to me

**Comment by veracioux** [comment]
@bhantsi Thanks for helping out.

**Comment by veracioux** [comment]
@bhantsi No. The basic git tutorial goes to the `git` file that is already in the repo. The other tutorial files you can name as `git-branch`, `git-rebase`, etc. The tutorials should be [tuterm](https://github.com/veracioux/tuterm) scripts. Please look at the documentation of that project (especially the man page) to learn the syntax of a tuterm script.
